### PR TITLE
 LXD Remove container retries on failure

### PR DESCRIPTION
**Pull request by SimonRichardson**
## Description of change

LXD sometimes (read as a lot) fails to clean up some containers,
and retrying helps fix the issues.

## QA steps

To exacerbate this more, it's ideal to have LXD setup with ZFS.

```
juju bootstrap lxd test
juju deploy redis
```

### CODE REVIEW

File: container/lxd/container.go [hunk 1]
```diff
@@ -339,6 +357,13 @@ func (s *Server) WriteContainer(c *Container) error {
 	return nil
 }
 
+func (s *Server) Clock() clock.Clock {
+	if s.clock == nil {
+		return clock.WallClock
```
Comment (manadart) [inline container/lxd/container.go]: Do we want to assign `WallClock` to `s.Clock` instead?

File: container/lxd/container.go [hunk 2]
```diff
@@ -318,12 +321,27 @@ func (s *Server) RemoveContainer(name string) error {
 		}
 	}
 
-	op, err := s.DeleteContainer(name)
-	if err != nil {
-		return errors.Trace(err)
+	retryArgs := retry.CallArgs{
```
Comment (manadart) [inline container/lxd/container.go]: Maybe a comment regarding the upstream LXD behaviour/advice?

**Comment by manadart** [comment]
Oh, BTW. This should probably go to 2.5, then forward. Helpful to trigger all the builds too :)

File: container/lxd/container.go (hunk 2 above)
Comment (SimonRichardson) [reply container/lxd/container.go]: Agreed

**Comment by SimonRichardson** [comment]
$$merge$$

**Pull request merged**